### PR TITLE
Rework sliding bank detail banner

### PR DESCRIPTION
**Pull request by KaiNissen**
This PR is a complete overhaul of the bank detail banner that is popping up after a certain idle time on the donation form.

The related bug is described in wmde/fundraising#1282, the task is tracked in [phab:162444](https://phabricator.wikimedia.org/T162444).

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Such CSS

+0.1

File: app/templates/Bank_Detail_Banner.html.twig [hunk 1]
```diff
@@ -1,131 +1,79 @@
...
-$( document ).ready( function() {
-  timer = window.setTimeout( function() {
-    $( '#bank-detail-banner' ).fadeIn();
-  }, 5000 );
-
-  $( 'input' ).focus( function() {
-    $( '#bank-detail-banner' ).fadeOut();
-    clearTimeout( timer );
-  } );
+	var timer, bankDetailBannerClosed = false;
```
Comment (wiese) [inline app/templates/Bank_Detail_Banner.html.twig]: Mind if we move these inside the .ready callback to make absolutely obvious these are not global?

File: app/templates/Bank_Detail_Banner.html.twig [hunk 2]
```diff
@@ -1,131 +1,79 @@
...
 
-  /* switch banner pages */
-  $( '.switch-button' ).click( function() {
-    $( '#bank-details' ).slideToggle();
-  } );
-} );
+		/* switch banner pages */
+		$( '.switch-button' ).click( function() {
+			var contentWrapper = $( '.content-wrapper' ),
+				newValue = contentWrapper.css( 'top' ) === '0px'
```
Comment (wiese) [inline app/templates/Bank_Detail_Banner.html.twig]: `jQuery.css()`
"When a number is passed as the value, jQuery will convert it to a string and add px to the end of that string. _If the property requires units other than px, convert the value to a string and add the appropriate units before calling the method._" (emphasis mine) TIL

File: app/templates/Bank_Detail_Banner.html.twig [hunk 3]
```diff
@@ -1,131 +1,79 @@
...
+                    ? '-4.4em' : '0em';
+			contentWrapper.animate( { top: newValue } );
```
Comment (wiese) [inline app/templates/Bank_Detail_Banner.html.twig]: `jQuery.animate()`
"The units em and % can be specified where applicable."

File: app/templates/Base_Layout.html.twig [hunk 4]
```diff
@@ -78,6 +78,10 @@
             {% endblock %}
         </div>
     </footer>
+
+    <!-- Banner -->
+    {% block banner %}
+    {% endblock %}
 </div>
```
Comment (wiese) [inline app/templates/Base_Layout.html.twig]: Yes!

File: app/templates/Bank_Detail_Banner.html.twig (hunk 1 above)
Comment (wiese) [inline app/templates/Bank_Detail_Banner.html.twig]: @KaiNissen Would love to hear your opinion before merging...

Comment (KaiNissen) [inline app/templates/Bank_Detail_Banner.html.twig]: No, I'm fine with that. It's unlikely that we need to access them from any other context.

**Pull request merged**